pkg/tsdb/elasticsearch/elasticsearch.go

```diff
@@ -122,16 +122,24 @@ func newInstanceSettings(httpClientProvider httpclient.Provider) datasource.Inst
 			maxConcurrentShardRequests = 256
 		}
 
-		includeFrozen, ok := jsonData["includeFrozen"].(bool)
+		includeFrozenFromJson, ok := jsonData["includeFrozen"].(bool)
 		if !ok {
-			includeFrozen = false
+			includeFrozenFromJson = false
 		}
 
+		// this field is not editable anymore,
+		// we have it because the following combination of json-fields:
+		// - xpack=false
+		// - includefrozen=true
+		// should resolve to `false`,
+		// to be 100% backward-compatible.
 		xpack, ok := jsonData["xpack"].(bool)
 		if !ok {
 			xpack = false
 		}
```

Comment:
I am okay with going with this approach. 

But my personal opinion is, that being 100% backward compatible in this (in my opinion) edge case/unclear UX situation makes code quite more messy and we are referencing field that will not exist anymore at couple of different places - here, data source config, datasource.ts. So in my opinion, it would be better to add a breaking change for this very edge case and move forward with cleaner and clearer code.

Reply:
i agree 👍 . for now i'll update this PR to be much smaller, to just hide the xpack-toggle in datasource-config, and we can do the breaking-change later.